nameserver/zone.go

```diff
@@ -121,6 +122,37 @@ func (zone *ZoneDb) AddRecord(ident string, name string, ip net.IP) error {
 	return nil
 }
 
+func (zone *ZoneDb) DeleteRecords(ident *string, ip *net.IP, name *string) (count int) {
+	zone.mx.Lock()
+	defer zone.mx.Unlock()
+
+	var fqdn *string
+	if name != nil {
+		temp := dns.Fqdn(*name)
+		fqdn = &temp
+	}
+
+	match := func(r dbRecord) bool {
+		return (ident == nil || r.Ident == *ident) &&
+			(ip == nil || r.IP.Equal(*ip)) &&
+			(fqdn == nil || r.Name == *fqdn)
+	}
+
+	w := 0 // write index
+
+	for _, r := range zone.recs {
+		if !match(r) {
+			zone.recs[w] = r
+			w++
+		}
+	}
+
+	count = len(zone.recs) - w
+	zone.recs = zone.recs[:w]
+
+	return
+}
+
```

Comment:
I think it would be a good idea to use that predicate for the internal implementation, but not for the API. I also I think that we will not really need that level of flexibility. We will probably need:

* to remove all records for an ident (ident, *, *) (eg, a container has been removed)
* to remove all records for a name in a ident (ident, name, *)
* to remove a particular record (ident, name, ip)

Can you guys think of any other cases?